modules/imgproc/src/smooth.cpp

```diff
@@ -3182,50 +3224,52 @@ template <>
 void vlineSmooth5N<uint8_t, ufixedpoint16>(const ufixedpoint16* const * src, const ufixedpoint16* m, int, uint8_t* dst, int len)
 {
     int i = 0;
-    static const v_int16x8 v_128 = v_reinterpret_as_s16(v_setall_u16((uint16_t)1 << 15));
-    v_int32x4 v_128_4 = v_setall_s32(128 << 16);
-    if (len > 7)
+#if CV_SIMD
+    static const v_int16 v_128 = v_reinterpret_as_s16(vx_setall_u16((uint16_t)1 << 15));
+    v_int32 v_128_4 = vx_setall_s32(128 << 16);
+    const int VECSZ = v_uint16::nlanes;
+    if (len >= VECSZ)
     {
         ufixedpoint32 val[] = { (m[0] + m[1] + m[2] + m[3] + m[4]) * ufixedpoint16((uint8_t)128) };
-        v_128_4 = v_setall_s32(*((int32_t*)val));
-    }
-    v_int16x8 v_mul01 = v_reinterpret_as_s16(v_setall_u32(*((uint32_t*)m)));
-    v_int16x8 v_mul23 = v_reinterpret_as_s16(v_setall_u32(*((uint32_t*)(m + 2))));
-    v_int16x8 v_mul4 = v_reinterpret_as_s16(v_setall_u16(*((uint16_t*)(m + 4))));
-    for (; i <= len - 32; i += 32)
-    {
-        v_int16x8 v_src00, v_src10, v_src01, v_src11, v_src02, v_src12, v_src03, v_src13;
-        v_int16x8 v_tmp0, v_tmp1;
-
-        v_src00 = v_load((int16_t*)(src[0]) + i);
-        v_src01 = v_load((int16_t*)(src[0]) + i + 8);
-        v_src02 = v_load((int16_t*)(src[0]) + i + 16);
-        v_src03 = v_load((int16_t*)(src[0]) + i + 24);
-        v_src10 = v_load((int16_t*)(src[1]) + i);
-        v_src11 = v_load((int16_t*)(src[1]) + i + 8);
-        v_src12 = v_load((int16_t*)(src[1]) + i + 16);
-        v_src13 = v_load((int16_t*)(src[1]) + i + 24);
+        v_128_4 = vx_setall_s32(*((int32_t*)val));
```

Comment:
This variable is used inside of loop only.
Loop will not run if "len < 4*VECSZ".
Is initial value from line 3229 used anywhere?
> v_int32 v_128_4 = vx_setall_s32(128 << 16);

May be add minimal length "if" check for all optimization block right after `#if CV_SIMD`?